### PR TITLE
Avoid error in compareTo if Nullable field is null

### PR DESCRIPTION
**Pull request by corinaminer**
Can't use `thenComparing(AclLineMatchExpr::getDescription)` because `_description` is Nullable and `thenComparing()` throws exceptions for null arguments.

### CODE REVIEW

**Comment by dhalperi** [comment]
Review status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @corinaminer, @anothermattbrown, @dhalperi, and @arifogel)

---

*[projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/acl/AclLineMatchExpr.java, line 40 at r1](https://reviewable.io/reviews/batfish/batfish/1708#-LFP4OAqDB7INgrTqk7g:-LFP4OArC91NKe2-1dS8:b-1rylrn) ([raw file](https://github.com/batfish/batfish/blob/4d8c61689c8a3923c122f896bedd4fe1868b4b81/projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/acl/AclLineMatchExpr.java#L40)):*
> ```Java
>       return Objects.compare(
>           _description, o._description, Comparator.nullsFirst(Comparator.naturalOrder()));
>     }
> ```

Can you use either `.nullsFirst()` or `.nullsLast()` instead of this?

---

*[projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/acl/AclLineMatchExpr.java, line 42 at r1](https://reviewable.io/reviews/batfish/batfish/1708#-LFP4SsD1OUV48z8dAjT:-LFP4SsE9F7e6Hik4Yh8:bqjkx7j) ([raw file](https://github.com/batfish/batfish/blob/4d8c61689c8a3923c122f896bedd4fe1868b4b81/projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/acl/AclLineMatchExpr.java#L42)):*
> ```Java
>     }
>     return compareWithoutDescription;
>   }
> ```

Missing new unit test.

---


*Comments from [Reviewable](https://reviewable.io/reviews/batfish/batfish/1708)*

**Comment by dhalperi** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed, 2 unresolved discussions (waiting on @corinaminer, @anothermattbrown, and @arifogel)

---



*Comments from [Reviewable](https://reviewable.io/reviews/batfish/batfish/1708)*

**Comment by corinaminer** [comment]
Review status: 0 of 2 files reviewed, all discussions resolved (waiting on @dhalperi, @anothermattbrown, and @arifogel)

---

*[projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/acl/AclLineMatchExpr.java, line 40 at r1](https://reviewable.io/reviews/batfish/batfish/1708#-LFP4OAqDB7INgrTqk7g:-LFP9PTY7EvUiFtOMaV8:b-896fix) ([raw file](https://github.com/batfish/batfish/blob/4d8c61689c8a3923c122f896bedd4fe1868b4b81/projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/acl/AclLineMatchExpr.java#L40)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

Can you use either `.nullsFirst()` or `.nullsLast()` instead of this?
</blockquote></details>

Done.

---

*[projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/acl/AclLineMatchExpr.java, line 42 at r1](https://reviewable.io/reviews/batfish/batfish/1708#-LFP4SsD1OUV48z8dAjT:-LFP9QwSBYw4BnKPXbgE:b-896fix) ([raw file](https://github.com/batfish/batfish/blob/4d8c61689c8a3923c122f896bedd4fe1868b4b81/projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/acl/AclLineMatchExpr.java#L42)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

Missing new unit test.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/batfish/batfish/1708)*

**Comment by dhalperi** [comment]
Review status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @dhalperi, @corinaminer, @anothermattbrown, and @arifogel)

---

*[projects/batfish-common-protocol/src/test/java/org/batfish/datamodel/acl/AclLineMatchExprTest.java, line 10 at r2](https://reviewable.io/reviews/batfish/batfish/1708#-LFPA3M00goclZNbHQyG:-LFPA3M00goclZNbHQyH:b-4czzmt) ([raw file](https://github.com/batfish/batfish/blob/29a6940ecffa8baad864d8e505bd0f2a2a1d1fb6/projects/batfish-common-protocol/src/test/java/org/batfish/datamodel/acl/AclLineMatchExprTest.java#L10)):*
> ```Java
> 
> public class AclLineMatchExprTest {
> ```

Fine as-is, but here's a simpler/more rigorous way to write this test:

```
private static final AclLineMatchExpr[] EXPRS = {
    // false
    // true
    // ... whatever/
};

@Test
public void testComapreTo() {
    for (int i = 0; i + 1 < EXPRS.lenght; ++i) {
        for (int j = i + 1; j < EXPRS.length; ++j) {
            assertThat(EXPRS[i], lessThan(EXPRS[j]));
        }
    }
}
```

or something close to that.

Basically, simpler and more readable code that actually tests more.,,

---


*Comments from [Reviewable](https://reviewable.io/reviews/batfish/batfish/1708)*

**Comment by dhalperi** [comment]
Reviewed 2 of 2 files at r2.
Review status: all files reviewed, 1 unresolved discussion (waiting on @corinaminer, @anothermattbrown, and @arifogel)

---



*Comments from [Reviewable](https://reviewable.io/reviews/batfish/batfish/1708)*

**Comment by dhalperi** [comment]
:lgtm:

---

Review status: all files reviewed, 1 unresolved discussion (waiting on @corinaminer, @anothermattbrown, and @arifogel)

---



*Comments from [Reviewable](https://reviewable.io/reviews/batfish/batfish/1708#-:-LFPAbF29ngsey52K1F7:bnfp4nl)*

**Comment by corinaminer** [comment]
Review status: 1 of 2 files reviewed, all discussions resolved (waiting on @dhalperi, @anothermattbrown, and @arifogel)

---

*[projects/batfish-common-protocol/src/test/java/org/batfish/datamodel/acl/AclLineMatchExprTest.java, line 10 at r2](https://reviewable.io/reviews/batfish/batfish/1708#-LFPA3M00goclZNbHQyG:-LFPN6q89Yv_G9YhylwR:bd882fv) ([raw file](https://github.com/batfish/batfish/blob/29a6940ecffa8baad864d8e505bd0f2a2a1d1fb6/projects/batfish-common-protocol/src/test/java/org/batfish/datamodel/acl/AclLineMatchExprTest.java#L10)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

Fine as-is, but here's a simpler/more rigorous way to write this test:

```
private static final AclLineMatchExpr[] EXPRS = {
    // false
    // true
    // ... whatever/
};

@Test
public void testComapreTo() {
    for (int i = 0; i + 1 < EXPRS.lenght; ++i) {
        for (int j = i + 1; j < EXPRS.length; ++j) {
            assertThat(EXPRS[i], lessThan(EXPRS[j]));
        }
    }
}
```

or something close to that.

Basically, simpler and more readable code that actually tests more.,,
</blockquote></details>

nice. ended up being a bit unwieldy/redundant/unintuitive if i included testing for equal values in the same setup, so i kept those separate

---


*Comments from [Reviewable](https://reviewable.io/reviews/batfish/batfish/1708)*

**Comment by dhalperi** [comment]
Reviewed 1 of 1 files at r3.
Review status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @anothermattbrown and @arifogel)

---



*Comments from [Reviewable](https://reviewable.io/reviews/batfish/batfish/1708)*
